src/react/hooks/__tests__/useLoadableQuery.test.tsx

```diff
@@ -0,0 +1,4848 @@
+    { client }
+  );
+
+  await act(() => user.click(screen.getByText("Load query")));
+
+  // initial render
+  await Profiler.takeRender();
+  // load query
+  await Profiler.takeRender();
+
+  {
+    const { snapshot, renderedComponents } = await Profiler.takeRender();
+
+    expect(renderedComponents).toStrictEqual([ReadQueryHook]);
+    expect(snapshot.result).toEqual({
+      data: { greeting: "Hello" },
+      error: undefined,
+      networkStatus: NetworkStatus.ready,
+    });
+  }
+
+  client.writeQuery({
+    query,
+    data: { greeting: "Updated Hello" },
+  });
+
+  {
+    const { snapshot, renderedComponents } = await Profiler.takeRender();
+
+    expect(renderedComponents).toStrictEqual([ReadQueryHook]);
+    expect(snapshot.result).toEqual({
+      data: { greeting: "Updated Hello" },
+      error: undefined,
+      networkStatus: NetworkStatus.ready,
+    });
+  }
+
+  await expect(Profiler).not.toRerender();
+});
+
+it("applies `errorPolicy` on next fetch when it changes between renders", async () => {
+  const { query } = useSimpleQueryCase();
+
+  const mocks: MockedResponse<SimpleQueryData>[] = [
+    {
+      request: { query },
+      result: { data: { greeting: "Hello" } },
+    },
+    {
+      request: { query },
+      result: {
+        errors: [new GraphQLError("oops")],
+      },
+    },
+  ];
+
+  const Profiler = createDefaultProfiler<SimpleQueryData>();
+  const { SuspenseFallback, ReadQueryHook, ErrorBoundary, ErrorFallback } =
+    createDefaultProfiledComponents(Profiler);
+
+  function App() {
+    const [errorPolicy, setErrorPolicy] = useState<ErrorPolicy>("none");
+    const [loadQuery, queryRef, { refetch }] = useLoadableQuery(query, {
+      errorPolicy,
+    });
+
+    return (
+      <>
+        <button onClick={() => setErrorPolicy("all")}>
+          Change error policy
+        </button>
+        <button onClick={() => refetch()}>Refetch greeting</button>
+        <button onClick={() => loadQuery()}>Load query</button>
+        <Suspense fallback={<SuspenseFallback />}>
+          <ErrorBoundary>
+            {queryRef && <ReadQueryHook queryRef={queryRef} />}
+          </ErrorBoundary>
+        </Suspense>
+      </>
+    );
+  }
+
+  const { user } = renderWithMocks(<App />, {
+    mocks,
+    wrapper: ({ children }) => <Profiler>{children}</Profiler>,
+  });
+
+  await act(() => user.click(screen.getByText("Load query")));
+
+  // initial render
+  await Profiler.takeRender();
+  // load query
+  await Profiler.takeRender();
+
+  {
+    const { snapshot } = await Profiler.takeRender();
+
+    expect(snapshot.result).toEqual({
+      data: { greeting: "Hello" },
+      error: undefined,
+      networkStatus: NetworkStatus.ready,
+    });
+  }
+
+  await act(() => user.click(screen.getByText("Change error policy")));
+  await act(() => user.click(screen.getByText("Refetch greeting")));
+
+  // change error policy
+  await Profiler.takeRender();
+  // refetch
+  await Profiler.takeRender();
```

Comment:
To me, it would be more readable like this - even without the comments - WDYT?
```suggestion
  await act(() => user.click(screen.getByText("Change error policy")));
  await Profiler.takeRender();

  await act(() => user.click(screen.getByText("Refetch greeting")));
  await Profiler.takeRender();
```

Reply:
Love it

Reply:
[935ff7c](https://github.com/apollographql/apollo-client/pull/11300/commits/935ff7cf2e5d2ba03285628519f67c1073b659de)